### PR TITLE
[build] Update IMEX version

### PR DESCRIPTION
**Pull request by Menooker**
The IMEX upstream has rebased its LLVM version to a recent commit (early August). In this PR, we update the LLVM commit for IMEX build and the IMEX commit as well. The IMEX commit points to the newest main-branch commit of the repo https://github.com/intel/mlir-extensions
~For now, We don't need self-hosted IMEX fork.~ We still need a patch on IMEX at https://github.com/Menooker/mlir-extensions/pull/1 from @dchigarev . Currently still using a forked IMEX, but the patch seems a bug fix and might be able to be pushed upstream.

### CODE REVIEW

File: lib/gc/Transforms/GPU/LinalgToXeGPU.cpp [hunk 1]
```diff
@@ -83,6 +83,12 @@ struct TilesArray {
   SmallVector<SmallVector<Value>> tileMatrix;
 };
 
+static xegpu::TensorDescType getTensorDescType(llvm::ArrayRef<int64_t> shape,
```
Comment (Menooker) [inline lib/gc/Transforms/GPU/LinalgToXeGPU.cpp:86]: The IMEX has originally a single function like:

```c++
TensorDescType get(ArrayRef<int64_t> shape, Type type, int aaa = 1, bool bbb = true);
```

And they added another similar overload function:

```c++
TensorDescType get(ArrayRef<int64_t> shape, Type type, int aaa = 1, SomeOtherType bbb = {});
```

This makes the default arguments unusable, `xegpu::TensorDescType::get(shape, elementType)` will end up with ambitious overloading error.

**Comment by Menooker** [comment]
@leshikus @WangJialei-A It seems that updating llvm-imex commit hash will not trigger re-build of LLVM in CI? Would you please fix that?

**Comment by dchigarev** [comment]
> @leshikus @WangJialei-A It seems that updating llvm-imex commit hash will not trigger re-build of LLVM in CI? Would you please fix that?

I think you just simply need to add `cmake/llvm-version-imex.txt` [here](https://github.com/intel/graph-compiler/blob/afed96ca35abda7a54d3953dcf4f7fd6a4ad4ec7/.github/workflows/build-llvm.yml#L6-L9):

```diff
  push:
    paths:
      - cmake/llvm-version.txt
+     - cmake/llvm-version-imex.txt
      - .github/workflows/build-llvm.yml

```

File: scripts/compile.sh [hunk 2]
```diff
@@ -129,7 +129,7 @@ build_llvm() {
             git fetch --all
         fi
 
-        git checkout dev
+        git checkout main
```
Comment (dchigarev) [inline scripts/compile.sh:132]: let's make it consistent with our CI steps
```suggestion
        IMEX_HASH=$(cat cmake/imex-version.txt)
        git checkout ${IMEX_HASH}
```

**Comment by Menooker** [comment]
Current status: Test failed on f16_matmul_32x32.mlir
On my local test env, I got `out of resource` when calling `clFinish` to wait for the GPU queue, and this might indicate that GPU hangs after running the kernel. In the CI env, the execution succeeds with wrong results (all zero).

**Comment by dchigarev** [comment]
> Current status: Test failed on f16_matmul_32x32.mlir On my local test env, I got `out of resource` when calling `clFinish` to wait for the GPU queue, and this might indicate that GPU hangs after running the kernel. In the CI env, the execution succeeds with wrong results (all zero).

There was a patch in the `dev` branch of IMEX that fixes that (Menooker/mlir-extensions#1), it seems that it wasn't propagated to the new main branch

**Comment by Menooker** [comment]
> > Current status: Test failed on f16_matmul_32x32.mlir On my local test env, I got `out of resource` when calling `clFinish` to wait for the GPU queue, and this might indicate that GPU hangs after running the kernel. In the CI env, the execution succeeds with wrong results (all zero).
> 
> There was a patch in the `dev` branch of IMEX that fixes that ([Menooker/mlir-extensions#1](https://github.com/Menooker/mlir-extensions/pull/1)), it seems that it wasn't propagated to the new main branch

Many thanks, @dchigarev ! I have rebased that to the current HEAD of main branch of IMEX and it works. Are you interested in submitting your PR to IMEX upstream?

**Comment by dchigarev** [comment]
> Are you interested in submitting your PR to IMEX upstream?

Yes, will do

**Review by kurapov-peter — APPROVED** [review]

**Review by dchigarev — APPROVED** [review]

**Pull request merged**